### PR TITLE
ISPN-13578 Workaround for Elytron not using supplied providers

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-13578

### CODE REVIEW

File: server/runtime/src/main/java/org/infinispan/server/SecurityActions.java [hunk 1]
```diff
@@ -52,6 +53,17 @@ static Properties getSystemProperties() {
       return doPrivileged(System::getProperties);
    }
 
+   static void addSecurityProvider(Provider provider) {
+      doPrivileged(() -> {
+               if (java.security.Security.getProvider(provider.getName()) == null) {
+                  java.security
+                        .Security.insertProviderAt(provider, 1);
```
Comment (danberindei) [inline server/runtime/src/main/java/org/infinispan/server/SecurityActions.java:60]: Weird wrapping

**Review by danberindei — CHANGES_REQUESTED** [review]
I believe the `ServerConfigurationSerializerTest.testConfigurationSerialization` failures at 
https://ci.infinispan.org/job/Infinispan/job/PR-9742/1/ are related.

**Comment by danberindei** [comment]
There's also a thread leak failure caused by `DigestMechanismFactory` using a static `nonceManager` with `new ScheduledThreadPoolExecutor(1)` as an executor and `EmbeddedInfinispanServerDriver` disabling the `DigestMechanismFactory` shutdown:

https://ci.infinispan.org/job/Infinispan/job/PR-9742/1/testReport/junit/(root)/UNKNOWN/ThreadLeakChecker/

Tristan, maybe we should create an Elytron JIRA asking them to add a `DigestMechanismFactory` constructor that takes a custom executor?

**Review by pruivo — APPROVED** [review]

**Comment by tristantarrant** [comment]
My Digest hack broke the Digest tests :)

**Comment by pruivo** [comment]
@tristantarrant checkstyle error `EmbeddedInfinispanServerDriver.java:25:8: Unused import - org.wildfly.security.http.digest.DigestMechanismFactory. [UnusedImports]`

**Comment by tristantarrant** [comment]
Tests fixed

**Pull request merged**